src/main/java/com.lelek.springBoot/Sender.java

```diff
@@ -0,0 +1,67 @@
+package com.lelek.springBoot;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Scanner;
+
+@Slf4j
+@Setter
+@Service
+public class Sender extends Thread{
+
+    private boolean stop = false;
+
+    @Autowired
+    private JavaMailSender javaMailSender;
+
+    private List<MySimpleMailMessage> getAllMessages() throws IOException {
```

Comment:
I feel like this class is doing more than it should. As the name states, this component is responsible for sending the messages. Storage is a different thing. As we discussed, we want to have an ability to switch storage implementation in future: now it's file-based, but we want to support database. If you will stick to this approach this component will quickly become overloaded by functionality and hard to work with